src/main/java/org/graylog/integrations/aws/AWSClientBuilderUtil.java

```diff
@@ -0,0 +1,100 @@
+package org.graylog.integrations.aws;
+
+import com.google.common.base.Preconditions;
+import org.apache.commons.lang3.StringUtils;
+import org.graylog.integrations.aws.resources.requests.AWSRequest;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.awscore.client.builder.AwsClientBuilder;
+import software.amazon.awssdk.core.client.builder.SdkClientBuilder;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.cloudwatchlogs.CloudWatchLogsClient;
+import software.amazon.awssdk.services.cloudwatchlogs.CloudWatchLogsClientBuilder;
+import software.amazon.awssdk.services.iam.IamClient;
+import software.amazon.awssdk.services.iam.IamClientBuilder;
+import software.amazon.awssdk.services.kinesis.KinesisClient;
+import software.amazon.awssdk.services.kinesis.KinesisClientBuilder;
+
+import java.net.URI;
+
+/**
+ * Responsible for initializing and building AWS SDK clients. This logic is centralized in one place to ensure consistency amongst the
+ * clients and their initialization.
+ */
+public class AWSClientBuilderUtil {
+
+    // Non-instantiable Util class.
+    private AWSClientBuilderUtil() {
+
+    }
+
+    /**
+     * Initialize the builder with the appropriate authorization, region, and endpoints.
+     *
+     * @param builder  Any AWS client builder.
+     * @param endpoint See {@link SdkClientBuilder#endpointOverride(java.net.URI)} javadoc.
+     * @param region   The region to specify on the client.
+     */
+    public static void initializeBuilder(AwsClientBuilder builder, String endpoint, Region region, AwsCredentialsProvider credentialsProvider) {
+        builder.region(region);
```

Comment:
@ceruleancee Don't we have to perform null checks for endpoint or AWSCredentialProvider?

Reply:
@lingpri Great point. I agree this is a good idea to do. I should probably do that consistently in the future. I like how the following example does this check (I believe this is a mostly standard way in Graylog code). What do you think?

It it is ok with you, I would like to wait on making this improvement, so it can be done later in a smaller single PR.

https://github.com/Graylog2/graylog2-server/blob/6fdb71e6b99c6b9fcad7d1853a64dfa6384a7609/graylog2-server/src/main/java/org/graylog2/cluster/ClusterConfigServiceImpl.java#L75

Reply:
@danotorrey -  I will have to slow think about this . Thanks for providing a good example . Yeah , please wait on making this improvement.